concourse/replicator.py

```diff
@@ -266,10 +266,18 @@ def ok(self):
 
 
 class DefinitionDeployer:
-    def deploy(self, definition_descriptor, pipeline):
+    def deploy(self, definition_descriptor):
         raise NotImplementedError('subclasses must overwrite')
 
 
+class NoOpDeployer:
+    def deploy(self, definition_descriptor):
+        return DeployResult(
```

Comment:
nit: maybe we should at least let no-op deployed write a log entry, stating that it successfully ignored the request to do sth.?